components/core/tests/test-end_to_end.cpp

```diff
@@ -0,0 +1,121 @@
+
+// NOLINTNEXTLINE(readability-function-cognitive-complexity)
+TEMPLATE_TEST_CASE(
+        "clp-s_compression_and_extraction_no_floats",
+        "[clp-s][end-to-end]",
+        four_byte_encoded_variable_t,
+        eight_byte_encoded_variable_t
+) {
+    std::filesystem::remove_all("test-end-to-end-archive");
+    std::filesystem::remove_all("test-end-to-end-out");
+    std::filesystem::remove("test-end-to-end_sorted.json");
+    std::filesystem::remove("diff_out.txt");
+
+    std::filesystem::create_directory("test-end-to-end-archive");
+    REQUIRE(std::filesystem::is_directory("test-end-to-end-archive"));
+
+    clp_s::JsonParserOption parser_option{};
+    parser_option.file_paths.push_back(get_test_input_local_path());
+    parser_option.archives_dir = "test-end-to-end-archive";
+    parser_option.target_encoded_size = cDefaultTargetEncodedSize;
+    parser_option.max_document_size = cDefaultMaxDocumentSize;
+    parser_option.min_table_size = cDefaultMinTableSize;
+    parser_option.compression_level = cDeaultCompressionLevel;
+    parser_option.print_archive_stats = cDefaultPrintArchiveStats;
+    parser_option.structurize_arrays = cDefaultStructurizeArrays;
+
+    clp_s::JsonParser parser(parser_option);
+    REQUIRE(parser.parse());
+    parser.store();
+
+    REQUIRE(false == std::filesystem::is_empty("test-end-to-end-archive"));
+
+    std::filesystem::create_directory("test-end-to-end-out");
+    REQUIRE(std::filesystem::is_directory("test-end-to-end-out"));
+
+    clp_s::JsonConstructorOption constructor_option{};
+    constructor_option.output_dir = "test-end-to-end-out";
+    constructor_option.ordered = false;
+    constructor_option.archives_dir = parser_option.archives_dir;
+    constructor_option.ordered_chunk_size = 0;
+    for (auto const& entry : std::filesystem::directory_iterator(constructor_option.archives_dir)) {
+        if (false == entry.is_directory()) {
+            // Skip non-directories
+            continue;
+        }
+
+        constructor_option.archive_id = entry.path().filename();
+        clp_s::JsonConstructor constructor(constructor_option);
+        constructor.store();
+    }
+
+    REQUIRE(std::filesystem::exists("test-end-to-end-out/original"));
+
+    int result = std::system("command -v jq >/dev/null 2>&1");
+    REQUIRE(0 == result);
+    result = std::system(
+            "jq -S -c '.' test-end-to-end-out/original | sort > test-end-to-end_sorted.json"
+    );
+    REQUIRE(0 == result);
+
+    REQUIRE(false == std::filesystem::is_empty("test-end-to-end_sorted.json"));
+
+    result = std::system("command -v diff >/dev/null 2>&1");
+    REQUIRE(0 == result);
+    std::string const command = "diff -u test-end-to-end_sorted.json " + get_test_input_local_path()
+                                + " > diff_out.txt";
+    result = std::system(command.c_str());
+    REQUIRE((0 == WEXITSTATUS(result) || 1 == WEXITSTATUS(result)));
+
+    REQUIRE(std::filesystem::is_empty("diff_out.txt"));
```

Comment:
I don't like producing side effects from tests, though I do see why you want to leave this diff file around. Even still, we should at least make sure this file gets deleted when the test is successful.

Reply:
Actually on second thought it's probably best to just make sure everything gets deleted via fixture (including this diff) regardless of success or failure just to keep things simple. It should be relatively easy to reproduce failures by just looking at the test and running the commands.